pkg/block/azure/adapter.go

```diff
@@ -568,7 +568,7 @@ func (a *Adapter) CompleteMultiPartUpload(ctx context.Context, obj block.ObjectP
 
 func (a *Adapter) GetStorageNamespaceInfo() block.StorageNamespaceInfo {
 	info := block.DefaultStorageNamespaceInfo(block.BlockstoreTypeAzure)
-	info.ValidityRegex = `^https?://`
+	info.ValidityRegex = `^https?:\/\/[a-z,0-9]+\.blob\.core\.windows\.net`
```

Comment:
1. You are using ` so you don't need to escape / just the regex wildcards
2. Check with @guy-har about this one as far as I remember we may kept verification to minimal because of multiple options of storage namespace in azure

Reply:
@guy-har can you please comment.
AFAIU this is the only blob subdomain: https://learn.microsoft.com/en-us/azure/security/fundamentals/azure-domains
(It is also the only one we currently support)

Reply:
Don't remember the exact reason, I checked now I noticed they support custom domains, but I'm not sure we can support them. So I believe it's fine, If we do have the information maybe we should document